extensions/resource-deployment/src/ui/deployClusterWizard/pages/azureSettingsPage.ts

```diff
@@ -133,7 +133,8 @@ export class AzureSettingsPage extends WizardPageBase<DeployClusterWizard> {
 					self.wizard.registerDisposable(disposable);
 				},
 				onNewInputComponentCreated: (name: string, inputComponentInfo: InputComponentInfo): void => {
-					this.inputComponents[name] = { component: inputComponentInfo.component };
+					self.inputComponents[name] = inputComponentInfo;
```

Comment:
So why do we even need the inputComponents array anymore if it's just going to be kept up to date with wizard.inputComponents? Seems like we only need the one

Reply:
Maybe, but I do not want to expand the scope of those changes here for a quick bug fix. Using the page's inputComponents, the model values get updated at end of each page which gets potentially used in other pages. We will have to track those usages down. I can eliminate the page-specific inputComponents and update the model values at the end of each page using the inputComponents from the wizard. That will eliminate the 2nd copy without much risk with the only down side being that we are populating model values after 2nd page we are re-populating the already populated values from 1st page.


As I said before we can definitely do a lot of cleanup in the BDC code. An alternative would be to do the absolutely minimal fix of just making the change to modelviewutils.

Reply:
Sounds good. I agree going through and cleaning this up would be nice but that can be done later if needed.